include/Contact.php

```diff
@@ -534,19 +534,20 @@ function get_contact($url, $uid = 0, $no_update = false) {
 	$data = array();
 	$contact_id = 0;
 
-	// We first try the addr (nick@server.tld)
+	// We first try the nurl (http://server.tld/nick), most common case
 	$contacts = q("SELECT `id`, `avatar-date` FROM `contact`
-				WHERE `addr` = '%s'
-				AND `uid` = %d",
-		dbesc($url),
-		intval($uid));
+					WHERE `nurl` = '%s'
+					AND `uid` = %d",
```

Comment:
The first query had a `LIMIT 2`, and the case where multiple contacts are found for a single `uid` is covered later in the code. Basically, there was no reason to keep it.